src/Identity/UI/src/Areas/Identity/Pages/V3/Account/RegisterConfirmation.cshtml.cs

```diff
@@ -0,0 +1,90 @@
+
+using System;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.RazorPages;
+using Microsoft.Extensions.Hosting;
+
+namespace Microsoft.AspNetCore.Identity.UI.V3.Pages.Account.Internal
+
+{
+    /// <summary>
+    ///     This API supports the ASP.NET Core Identity default UI infrastructure and is not intended to be used
+    ///     directly from your code. This API may change or be removed in future releases.
+    /// </summary>
+    [AllowAnonymous]
+    [IdentityDefaultUI(typeof(RegisterConfirmationModel<>))]
+    public class RegisterConfirmationModel : PageModel
+    {
+        /// <summary>
+        ///     This API supports the ASP.NET Core Identity default UI infrastructure and is not intended to be used
+        ///     directly from your code. This API may change or be removed in future releases.
+        /// </summary>
+        public string Email { get; set; }
+
+        /// <summary>
+        ///     This API supports the ASP.NET Core Identity default UI infrastructure and is not intended to be used
+        ///     directly from your code. This API may change or be removed in future releases.
+        /// </summary>
+        public bool IsDevelopment { get; set; }
+
+        /// <summary>
+        ///     This API supports the ASP.NET Core Identity default UI infrastructure and is not intended to be used
+        ///     directly from your code. This API may change or be removed in future releases.
+        /// </summary>
+        public string EmailConfirmationUrl { get; set; }
+
+        /// <summary>
+        ///     This API supports the ASP.NET Core Identity default UI infrastructure and is not intended to be used
+        ///     directly from your code. This API may change or be removed in future releases.
+        /// </summary>
+        public virtual Task<IActionResult> OnGetAsync(string email) => throw new NotImplementedException();
+    }
+
+    internal class RegisterConfirmationModel<TUser> : RegisterConfirmationModel where TUser : class
+    {
+        private readonly UserManager<TUser> _userManager;
+        private readonly IWebHostEnvironment _hostingEnv;
+
+        public RegisterConfirmationModel(UserManager<TUser> userManager, IWebHostEnvironment hostingEnv)
+        {
+            _userManager = userManager;
+            _hostingEnv = hostingEnv;
+        }
+
+        public override async Task<IActionResult> OnGetAsync(string email)
+        {
+            if (email == null)
+            {
+                return RedirectToPage("/Index");
+            }
+
+            var user = await _userManager.FindByEmailAsync(email);
+            if (user == null)
+            {
+                return NotFound($"Unable to load user with email '{email}'.");
+            }
+
+            Email = email;
+            IsDevelopment = _hostingEnv.IsDevelopment();
```

Comment:
I don't think we do this in any part of ASP.NET (other than the the old metapackage) and we try to avoid this. (I tried to do this in the past and got smacked for it). The pattern that we follow is to flow this through config or other means. 

If I can provide an alternative suggestion, you can remove the property completely. You can have an if on the view like as follows
```csharp
if(ViewData["EnableConfirmationLink"])
{
    // Display the link
}
```
and you can drop a _ViewStart.cshtml (on the templates) under the Account folder to enable it as follows
```razor
@inject IHostingEnvironment HostingEnvironment;
ViewData["EnableConfirmationLink"] = HostingEnvironment.IsDevelopment
```
This way:
* The user has direct control of when this is enabled.
* We don't depend on hosting environment within the core libraries.

Reply:
I can just switch to checking for the no-op default email sender instead to control whether to display the code

Reply:
Hmm, you don't want that to happen in prod, do you?

Reply:
It will poison your data and it will be a mess.

Reply:
We'll go over this with @blowdart at triage this week and hash out this bit then

Reply:
The new way looks definitely better, but I would consider putting something on the template viewstart to make sure this never shows up in prod.